include/arch/zynq7000.h

```diff
@@ -44,7 +52,7 @@ enum {
 
 typedef struct {
 	enum { pctl_set = 0, pctl_get } action;
-	enum { pctl_ambaclock = 0, pctl_devclock, pctl_mioclock, pctl_mio, pctl_reboot } type;
+	enum { pctl_ambaclock = 0, pctl_devclock, pctl_mioclock, pctl_mio, pctl_reset, pctl_reboot } type;
```

Comment:
Thanks for spotting the ABI breaking change. I agree that the new pctl should be added at the end of the enum list, however it shouldn't be a problem for now since `zynq7000` platform development is still at an early stage. I'd like to keep `pctl_reboot` as the last one in order to remain consistent with the other platforms.

I've changed the new pctl name to `pctl_devreset`. @agkaminski and @HBuczynski aggred that it's a more meaningful name.

Reply:
I'm not blocking this particular PR but I would like to spur generic discussion. IMHO the rule of thumb should be - **don't beak the ABI if it's not absolutely needed** (additive changes are OK if kernel handles unknown (future) request in a sane way). If breaking ABI change is needed - mark commit as **breaking ABI change and describe it clearly** (every ABI change between OS releases will need to be described clearly - we need to introduce at least **some** traceability).

Maybe our approach to ABI definition is too fragile to maintain and we should rethink it? :thinking: 

The argument *I'm breaking the ABI but it shouldn't be a problem* is not good enough if we're hoping external developers would start using our OS :).